daemon/src/bin/fuzzer.rs

```diff
@@ -33,6 +33,8 @@ fn initialize_project() -> (temp_dir::TempDir, PathBuf) {
     (dir, file)
 }
 
+/* TODO: consider moving fuzzer to separate crate to avoid adding test code to ethersync binary. */
+/* TODO: consider using cargo-fuzz and exposing a stdin interface to avoid network overhead. */
```

Comment:
So the reason I made this comment is because I believe the term "fuzz" has a relatively specific meaning in cargo, including special support with `cargo fuzz`: https://rust-fuzz.github.io/book/cargo-fuzz.html. We have a separate subcrate for fuzzing in the `zip` crate (https://github.com/zip-rs/zip2/blob/af33ed343da5769e0587d638e435d67151990f10/fuzz/Cargo.toml#L8-L9), because it often requires different dependencies, and `cargo fuzz` links the binary against clang's ASan (which causes errors if it's in the same workspace as our `zip` library). I have contributed some changes to `zip`'s fuzzing -- I would *love* to do the same here (in a separate change).

In https://github.com/ethersync/ethersync/compare/main...cosmicexplorer:use-cargo-test-for-integration-tests?expand=1 (which I am going to move into an integration test before posting as a PR), I rename the binary from `fuzzer.rs`, because I thought it would be useful to make clear it's not using the *cargo fuzz* framework. However, I think that change is actually not necessary, since it definitely is still performing a sort of fuzzing, so I will change its name back before posting the integration test PR.

Reply:
Broken out in #130.